inc/datahandlers/login.php

```diff
@@ -109,7 +109,7 @@ function verify_captcha()
 				return true;
 			}
 		}
-		else if($mybb->input['quick_login'] == 1 && $mybb->input['quick_password'] && $mybb->input['quick_username'])
+		else if(isset($mybb->input['quick_login']) && $mybb->input['quick_login'] == 1 && $mybb->input['quick_password'] && $mybb->input['quick_username'])
```

Comment:
You could probably use `!empty($mybb->input['quick_login'])` or $mybb->get_input() here.

Reply:
Try `$mybb->get_input('quick_login', \MyBB::INPUT_INT) === 1` instead.

Reply:
Thanks for your suggestions! Thinking it over, we only care about the truthiness of `quick_login`, not its numeric value. So, I prefer the `!empty` approach:

```suggestion
		else if(!empty($mybb->input['quick_login']) && $mybb->input['quick_password'] && $mybb->input['quick_username'])
```

Apart from that, for consistency’s sake we should check the other array keys, too, rather than assume that `quick_password` and `quick_username` must be set if `quick_login` is:

```suggestion
		else if(!empty($mybb->input['quick_login']) && !empty($mybb->input['quick_password']) && !empty($mybb->input['quick_username']))
```